### DISCUSSION TITLE
Is it possible to exclude components at runtime?

### DISCUSSION THREAD

**Discussion by nestle49** [nuxt/nuxt#16942] [Ideas]
Is it possible to exclude components at runtime?

I know about `.nuxtignore` allow to exclude components/directory from build

But is it possible at runtime?
Our project uses many themes (each includes directory with components) and each site use only one theme (got from api). User can select theme and api gives other theme. It means that build contains all components, even componens which never be used. Maybe it's possible, exclude components after change answer api?

**Comment by TheAlexLichter** [comment]
I am not aware that this is possible - but - you could lazy-load most of the theme components so that only the actually needed components are loaded for the user and downloaded JS size is kept small.
